### DISCUSSION TITLE
Poll Depdendency injection

### DISCUSSION THREAD

**Discussion by thebino** [AndroidDev-social/DodoForMastodon#54] [Polls]
As discussed in #51 we need to find a dependency injection solution for this project.
For details [follow this discussion](https://github.com/AndroidDev-social/MastodonCompose/discussions/51) before giving your vote.

**Comment by StylianosGakis** [comment]
If the decision is to go with Dagger. I'd like to suggest taking a look at [whetstone](https://github.com/deliveryhero/whetstone) too. It's basically like Hilt, while avoiding all the Kapt overhead by using [anvil](https://github.com/square/anvil) under the hood for all modules but the :app module.

**Comment by roshanrai06** [comment]
Things to Note here is Koin has an issue with SavedStateHandle in android  a comparison below

https://github.com/InsertKoinIO/koin/issues/1468

https://betterprogramming.pub/dagger-hilt-vs-koin-dependency-injection-for-jetpack-compose-apps-2e598f00aff

IMO iOS does not require much DI, For Android, we can use Hilt which is simple, and if it's required we can build our own DI or use Kotlin default for shared data and domain layer.
